docs/src/tutorial.jl

```diff
@@ -708,161 +707,87 @@ end
-)
-
-# This approach also works with the [`@multiagent`](@ref) possibility we discuss below.
-# `Union` types however also offer the unique possibility of utilizing fully the Julia's
-# [multiple dispatch system](https://docs.julialang.org/en/v1/manual/methods/).
-# Hence, we can use the same function name and add dispatch to it, such as:
-
-function dispatch_step!(agent::SchellingAgent, model)
+function agent_step!(agent::Schelling, model)
     ## stuff.
 end
 
-function dispatch_step!(agent::Politician, model)
+function agent_step!(agent::Politician, model)
     ## other stuff.
 end
 
-# and give `dispatch_step!` to the `agent_step!` keyword during model creation.
-
-# ## Multiple agent types with `@multiagent`
-
-# [`@multiagent`](@ref) is a macro, and hence not "native Julia" syntax like `Union`,
-# however it has been designed to be as similar to [`@agent`](@ref) as possible.
-# The syntax to use it is like so:
-
-@multiagent struct MultiSchelling{X}(GridAgent{2})
-    @subagent struct Civilian # can't re-define existing `Schelling` name
-        mood::Bool = false
-        group::Int
-    end
-    @subagent struct Governor{X} # can't redefine existing `Politician` name
-        group::Int
-        influence::X
-    end
-end
-
-# This macro created three names into scope:
-
-(MultiSchelling, Civilian, Governor)
-
-# however, only one of these names is an actual Julia type:
-
-fieldnames(MultiSchelling)
-
-# that contains all fields of all subtypes without duplication, while
+# and then passing
 
-fieldnames(Civilian)
+model = StandardABM(
+    Union{Schelling, Politician}, # type of agents
+    space; # space they live in
+    agent_step!
+)
 
-# doesn't have any fields. Instead,
-# you should think of `Civilian` and `Governor` as just convenience functions that have been
-# defined for you to "behave like" types. That's why we call these **kinds** and not
-# **types**.
+# ## Multiple agent types with `@sumtype`
 
-# E.g., you can initialize
+# By using `@sumtype` from `DynamicSumTypes.jl` it is possible to improve the 
+# computational performance of simulations requiring multiple types, while almost
+# everything works the same 
 
-civ = Civilian(; id = 2, pos = (2, 2), group = 2) # default `mood`
+using DynamicSumTypes
 
-# or
+@sumtype MultiSchelling(Schelling, Politician) <: AbstractAgent
 
-gov = Governor(; id = 3 , pos = (2, 2), group = 2, influence = 0.5)
+# Now when you create instances you will need to enclose them in `MultiSchelling`
 
-# exactly as if these were types made with [`@agent`](@ref).
-# These are all of type `MultiSchelling`
+p = MultiSchelling(Politician(model; pos = random_position(model), preferred_demographic = 1))
 
-typeof(gov)
+# agents are then all of type `MultiSchelling`
 
-# and hence you can't use `typeof` to differentiate them. But you can use
+typeof(p)
 
-kindof(gov)
+# and hence you can't use only `typeof` to differentiate them. But you can use
 
-# instead.
+typeof(variant(p))
```

Comment:
Right, okay, so we are in the situation that "You can't use `typeof` but you can use `kindof`." Now instead of `kind` of we have `typeof(variant`. Whether one or the other is more complex is up to debate. But this new approach doesn't appear to solve any problems or make things simpler.

Reply:
there is new also a `variantof` function also now